### PR TITLE
First attempt at a Supposition.jl-based test

### PR DESCRIPTION
**Pull request by bvrb**
Started during the Hackathon at JuliaCon Paris under the guidance of @Kolaru, here is my first attempt at adding a test using [Supposition.jl](https://github.com/Seelengrab/Supposition.jl) (c.p. https://github.com/JuliaIntervals/IntervalArithmetic.jl/issues/733).
To start simple, the goal was to convert an existing test with a hardcoded numerical value to one using randomly generated values from Supposition.jl instead. 
For no particular reason, we picked
https://github.com/JuliaIntervals/IntervalArithmetic.jl/blob/544c0005b96e70c026ebc306ebeeb7e7fed63274/test/interval_tests/numeric.jl#L175
which tests that for a degenerate interval with equal start and endpoints of type `Rational`, both the one and two argument syntax yield the same interval.

There were a few unforeseen complications, which I tried my best to find a solution for:

1. **Generation of Rationals**
  - As there is no ready-made Rational generator in Supposition.jl, after consulting @Seelengrab I opted to create a simple custom one, where both numerator and denominator are randomly drawn `Int`s, and the case `0//0` is explicitly thrown out as an invalid test case. 
  - Interesting edge cases are given by `1//0` and `-1//0` (corresponding `Inf`/`-Inf`), which will yield an ill-formed interval (see also https://github.com/JuliaIntervals/IntervalArithmetic.jl/issues/734). However, similar to the discussion in https://github.com/Seelengrab/Supposition.jl/discussions/62#discussioncomment-11862641, it is unlikely that a `0` is generated in the denominator for an `Int64`, so I had to use an `Int8` generator instead. As an alternative, one might instead add these cases explicitly to the regular tests.
  - There was another problem which took some time to track down, where sometimes an `OverflowError` would appear, especially when using `Int8`. The issue can be reproduced by
  ```julia
julia> num = Int8(3) # or any Int8 !=0 which is not a divisor of typemin(Int8)
3

julia> den = typemin(Int8)
-128

julia> num//den
ERROR: OverflowError: 0 - -128 overflowed for type Int8
Stacktrace:
 [1] throw_overflowerr_binaryop(op::Symbol, x::Int8, y::Int8)
   @ Base.Checked .\checked.jl:163
 [2] checked_sub
   @ .\checked.jl:232 [inlined]
 [3] checked_neg
   @ .\checked.jl:95 [inlined]
 [4] checked_den
   @ .\rational.jl:22 [inlined]
 [5] Rational
   @ .\rational.jl:44 [inlined]
 [6] Rational
   @ .\rational.jl:48 [inlined]
 [7] //(n::Int8, d::Int8)
   @ Base .\rational.jl:84
 [8] top-level scope
   @ REPL[66]:1
  ```
So I believe that upon construction of a Rational resulting in an overall negative value, a minus sign in the denominator is transferred to the numerator instead, and the sign of the former is flipped. However, if that value happens to be `typemin(Int8)`, there will be an overflow, since `typemax(Int8)==127`. To address this, I instead added `1` to the minimum value of the `Int` generator. Does that seem reasonable? 

2. **`1//0` and `-1//0` & Ill-formed intervals** 
  - The equality test failed whenever the result was an empty, ill-formed interval. As far as I understand the discussion in https://github.com/JuliaIntervals/IntervalArithmetic.jl/issues/734, this is the way it is supposed to be, as the behavior is similar to a `NaN`, and any Boolean comparison will yield `false`. 
  - For the purpose of the test, I therefore excluded those cases where the decorator is `ill`. 

Also, it might make sense to fix the random seed for reproducibility reasons, but I am not sure if this would go against the spirit of this fuzzing-based approach, which I am not too familiar with.

I apologize for the wall of text for this (at least in theory) rather simple addition, and would appreciate any feedback!

### CODE REVIEW

File: test/interval_tests/supposition/numeric.jl [hunk 1]
```diff
@@ -0,0 +1,21 @@
+using Test
+using IntervalArithmetic
+using Supposition, Supposition.Data
+
+@testset "Rational tests" begin
+    # Define number generators
+    #intgen = Data.Integers(typemin(Int)+1,typemax(Int)) # Don't allow den==typemin(Int) to avoid overflow
+    intgen = Data.Integers(typemin(Int8)+Int8(1),typemax(Int8)) # Don't allow den==typemin(Int8) to avoid overflow
```
Comment (Kolaru) [inline test/interval_tests/supposition/numeric.jl]: Does this still let the tests run with any integer type, or only Int8?

File: test/interval_tests/supposition/numeric.jl [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+        assume!(!(iszero(num) && iszero(den)))
+        return num // den
+    end
+
+    # Check properties
+    @check function degenerate_interval(r=rationalgen)
+        x = interval(r)
+        y = interval(r, r)
+        assume!(!any(decoration.((x,y)) .== ill)) # Exclude ill-formed intervals
+        isequal_interval(x, y)
```
[the remark this answers is not in the document]
Comment (bvrb) [reply test/interval_tests/supposition/numeric.jl]: Thanks! Changed it accordingly.

**Comment by Kolaru** [comment]
We can easily drop the 32bit tests yes, but that suboptimal (I don't know who's using 32 bit machines, but still).

@OlivierHnt Do you think that it's okay to drop 32bit machines from CI? Or should we wait for Suppositions.jl to fix before merging this PR?

**Comment by OlivierHnt** [comment]
Instead of dropping everything, can we just not run the tests using Suppositions.jl on 32 bit systems ?
So add something like:

```Julia
if Int != 32
    # run tests with Suppositions.jl
end
```
